src/portable/raspberrypi/rp2040/rp2040_usb.c

```diff
@@ -154,7 +154,14 @@ void _hw_endpoint_xfer_start(struct hw_endpoint *ep, uint8_t *buffer, uint16_t t
     ep->total_len = total_len;
     ep->len = 0;
     // FIXME: What if low speed
-    ep->transfer_size = total_len > 64 ? 64 : total_len;
+    if(ep->transfer_type == TUSB_XFER_ISOCHRONOUS)
+    {
+        ep->transfer_size = total_len > ep->wMaxPacketSize ? ep->wMaxPacketSize : total_len; 
+    }
+    else
+    {
+        ep->transfer_size = total_len > 64 ? 64 : total_len;
```

Comment:
I think this `else{}` could be merged to the above `if()` as well if 64 stands for ep size of non-iso here. Unless there is something I don't know of. @liamfraser may shed some light on this.

Reply:
64 is the max Non ISO EP size. We could potentially use a define for this.

Reply:
The ep size can actually be smaller for example interrupt ep can have max packet size of 8 or 16. Will we use the actual mps (8) here or 64 still is correct ? If 8 is needed then we should change it to wMaxPacketSize (like iso) instead of fixed 64.

Reply:
I think we should use wMaxPacketSize everywhere. I think that's what my "// FIXME: What if low speed" was meant for but I never got round to fixing it. I would like there to be a check that wMaxPacketSize is in a valid range for RP2040 somewhere though.

Reply:
Yeah, that is what I think as well. Will update pr accordingly. usbd already perform the endpoint size check already before passing it to dcd open, so I guess it is rather OK afterwards. https://github.com/hathach/tinyusb/blob/1bba2c0fc3bce05e9fbe4ff23dda30283d08574d/src/device/usbd.c#L1111

Reply:
Yeah, it changes recently to dynamically support FS/HS. It is mostly for  HS capable device to work properly with attached to let's say PICO FS host. Fullspeed device as default should have no issue, don't worry about low speed mode. It is virtually non-existent, no body is interested in it :smile: .